src/int.rs

```diff
@@ -0,0 +1,1409 @@
+//! This module contains an 256-bit signed integer implementation.
+
+use crate::errors::{ParseI256Error, TryFromBigIntError};
+use serde::{Deserialize, Serialize};
+use std::cmp;
+use std::convert::{TryFrom, TryInto};
+use std::fmt;
+use std::iter;
+use std::ops;
+use std::str;
+use std::{i128, i64, u64};
+use web3::contract::{self, tokens};
+use web3::types::U256;
+
+/// Compute the two's complement of a U256.
+fn twos_complement(u: U256) -> U256 {
+    let (twos_complement, _) = (!u).overflowing_add(U256::one());
+    twos_complement
+}
+
+/// Overflow handling that depends on whether or not the binary is built in
+/// debug mode.
+///
+/// # Panics
+///
+/// This function will panic on overflows in debug mode, and simply ignore the
+/// overflow in non-debug mode.
+#[inline(always)]
+fn handle_overflow<T>((result, overflow): (T, bool)) -> T {
+    #[cfg(debug_assertions)]
+    {
+        if overflow {
+            panic!("overflow");
+        }
+    }
+
+    let _ = overflow;
+    result
+}
+
+/// Enum to represent the sign of a 256-bit signed integer.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+enum Sign {
+    /// Greater than or equal to zero.
+    Positive,
+    /// Less than zero.
+    Negative,
+}
+
+impl Sign {
+    /// Computes the `Sign` given a signum.
+    fn from_signum64(sign: i64) -> Self {
+        match sign {
+            0 | 1 => Sign::Positive,
+            -1 => Sign::Negative,
+            _ => unreachable!(),
+        }
+    }
+}
+
+/// Little-endian 256-bit signed integer.
+#[derive(Clone, Copy, Default, Deserialize, Eq, Hash, Ord, PartialEq, Serialize)]
+#[serde(transparent)]
+pub struct I256(U256);
+
+impl I256 {
+    /// Maximum value.
+    pub const MAX: I256 = I256(U256([u64::MAX, u64::MAX, u64::MAX, i64::MAX as _]));
+
+    /// Minimum value.
+    pub const MIN: I256 = I256(U256([0, 0, 0, i64::MIN as _]));
+
+    /// Zero (additive identity) of this type.
+    pub const fn zero() -> Self {
+        I256(U256([0, 0, 0, 0]))
+    }
+
+    /// One (multiplicative identity) of this type.
+    pub const fn one() -> Self {
+        I256(U256([1, 0, 0, 0]))
+    }
+
+    /// Minus one (multiplicative inverse) of this type.
+    pub const fn minus_one() -> Self {
+        I256(U256([u64::MAX, u64::MAX, u64::MAX, u64::MAX]))
+    }
+
+    /// The maximum value which can be inhabited by this type.
+    pub const fn max_value() -> Self {
+        I256::MAX
+    }
+
+    /// The minimum value which can be inhabited by this type.
+    pub const fn min_value() -> Self {
+        I256::MIN
+    }
```

Comment:
Why introduce these if we already have the public constants?

Reply:
Just for consistency with primitive integer types where both the [constant exists](https://doc.rust-lang.org/std/u32/constant.MAX.html) as well as a [const method](https://doc.rust-lang.org/std/primitive.u32.html#method.max_value).